assets/js/editor-components/search-list-control/types.ts

```diff
@@ -77,9 +77,9 @@ export interface SearchListControlProps {
 	// Callback fired when the search field is used.
 	onSearch?: ( search: string ) => void;
 	// Callback to render each item in the selection list, allows any custom object-type rendering.
-	renderItem: ( args: renderItemArgs ) => JSX.Element;
+	renderItem?: ( args: renderItemArgs ) => JSX.Element;
```

Comment:
I changed this to optional because of this https://github.com/woocommerce/woocommerce-blocks/blob/fix/6496/assets/js/editor-components/search-list-control/search-list-control.tsx#L154.